### PR TITLE
Guidev

### PR DESCRIPTION
**Pull request by nwrogers**
First attempt at a GUI for Hotspot

### CODE REVIEW

File: src/mainwindow.cpp [hunk 1]
```diff
@@ -0,0 +1,126 @@
+/*
+  mainwindow.cpp
+
+  This file is part of Hotspot, the Qt GUI for performance analysis.
+
+  Copyright (C) 2016-2017 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+  Author: Milian Wolff <milian.wolff@kdab.com>
```
Comment (milianw) [inline src/mainwindow.cpp]: feel free to add yourself as Author here

File: src/mainwindow.cpp [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+#include "models/costmodel.h"
+#include "models/framedata.h"
+#include "parsers/perf/perfparser.h"
+
+MainWindow::MainWindow(QWidget *parent) :
+    QMainWindow(parent),
+    ui(new Ui::MainWindow),
+    m_bottomUpCostModel(new CostModel(this)),
+    m_parser(new PerfParser(this)),
+    m_LoadingResultsMovie(new QMovie(":/images/loading_icon.gif"))
```
Comment (milianw) [inline src/mainwindow.cpp]: I think this is overkill, please use a simply QProgressBar instead

File: src/mainwindow.ui [hunk 3]
```diff
@@ -0,0 +1,2174 @@
...
+    <x>0</x>
+    <y>0</y>
+    <width>809</width>
+    <height>406</height>
+   </rect>
+  </property>
+  <property name="windowTitle">
+   <string>MainWindow</string>
+  </property>
+  <property name="styleSheet">
```
Comment (milianw) [inline src/mainwindow.ui]: here and below: please undo the stylesheet changes, as it won't work nicely on dark color schemes

we can have one of our designers give us a nice spec to implement in the future. for now, let's keep it simple (and not as pretty)

File: src/mainwindow.ui [hunk 4]
```diff
@@ -0,0 +1,2174 @@
...
+           <height>56</height>
+          </size>
+         </property>
+         <property name="maximumSize">
+          <size>
+           <width>80</width>
+           <height>56</height>
+          </size>
+         </property>
+         <property name="palette">
```
Comment (milianw) [inline src/mainwindow.ui]: also don't meddle with the palette for now please

File: src/parsers/perf/perfparser.cpp [hunk 5]
```diff
@@ -33,14 +33,11 @@
 #include <QBuffer>
 #include <QDataStream>
 #include <QFileInfo>
-#include <QLoggingCategory>
```
Comment (milianw) [inline src/parsers/perf/perfparser.cpp]: you did the conflict resolution in the wrong direction here. please undo all changes to perfparser.cpp

**Comment by nwrogers** [comment]
The requested changes have been made and pushed

File: src/mainwindow.ui [hunk 6]
```diff
@@ -0,0 +1,716 @@
...
+        <item row="6" column="1">
+         <widget class="QLabel" name="loadingResultsErrorLabel">
+          <property name="font">
+           <font>
+            <family>Ubuntu</family>
+            <pointsize>11</pointsize>
+           </font>
+          </property>
+          <property name="styleSheet">
+           <string notr="true">color: rgb(255, 30, 30);</string>
```
Comment (milianw) [inline src/mainwindow.ui]: this is still hardcoded, but should not

File: src/mainwindow.ui [hunk 7]
```diff
@@ -0,0 +1,716 @@
...
+         </property>
+         <property name="maximumSize">
+          <size>
+           <width>80</width>
+           <height>56</height>
+          </size>
+         </property>
+         <property name="font">
+          <font>
+           <family>Ubuntu Mono</family>
```
Comment (milianw) [inline src/mainwindow.ui]: don't hardcode any fonts, use system default

File: src/mainwindow.ui [hunk 8]
```diff
@@ -0,0 +1,716 @@
...
+           <pointsize>11</pointsize>
```
Comment (milianw) [inline src/mainwindow.ui]: same for the font size

File: src/mainwindow.ui [hunk 9]
```diff
@@ -0,0 +1,716 @@
...
+         </property>
+         <property name="maximumSize">
+          <size>
+           <width>80</width>
+           <height>56</height>
+          </size>
+         </property>
+         <property name="font">
+          <font>
+           <family>Ubuntu Mono</family>
```
Comment (milianw) [inline src/mainwindow.ui]: dito

File: src/mainwindow.ui [hunk 10]
```diff
@@ -0,0 +1,716 @@
...
+           <size>
+            <width>295</width>
+            <height>20</height>
+           </size>
+          </property>
+         </spacer>
+        </item>
+        <item row="5" column="1">
+         <widget class="QLabel" name="loadingResultsLabel">
+          <property name="font">
```
Comment (milianw) [inline src/mainwindow.ui]: dito

File: src/mainwindow.ui [hunk 11]
```diff
@@ -0,0 +1,716 @@
...
+        </item>
+        <item row="1" column="0" colspan="3">
+         <widget class="QLabel" name="welcomeText">
+          <property name="maximumSize">
+           <size>
+            <width>729</width>
+            <height>16777215</height>
+           </size>
+          </property>
+          <property name="font">
```
Comment (milianw) [inline src/mainwindow.ui]: dito

File: src/mainwindow.ui [hunk 12]
```diff
@@ -0,0 +1,716 @@
...
+           <set>Qt::AlignCenter</set>
+          </property>
+          <property name="margin">
+           <number>0</number>
+          </property>
+         </widget>
+        </item>
+        <item row="6" column="1">
+         <widget class="QLabel" name="loadingResultsErrorLabel">
+          <property name="font">
```
Comment (milianw) [inline src/mainwindow.ui]: dito

File: src/mainwindow.ui [hunk 13]
```diff
@@ -0,0 +1,716 @@
...
+          </property>
+          <property name="styleSheet">
+           <string notr="true"/>
+          </property>
+          <property name="text">
+           <string>Open File</string>
+          </property>
+          <property name="icon">
+           <iconset>
+            <normalon>:/images/folder_icon.png</normalon>
```
Comment (milianw) [inline src/mainwindow.ui]: please instead use a system icon via the theme:

- open the command `kdialog --geticon` and find a fitting icon, here e.g. "document-open". there may be a kdialog alternative for GTK, or you can use https://specifications.freedesktop.org/icon-naming-spec/icon-naming-spec-latest.html - note that his is a shared standard and independent of KDE/Gnome/...
- in designer, expand the icon property and enter the icon name in the "Theme" row

File: src/mainwindow.ui [hunk 14]
```diff
@@ -0,0 +1,716 @@
...
+           <property name="toolTip">
+            <string>Application hotspot summary</string>
+           </property>
+           <attribute name="title">
+            <string>Summary</string>
+           </attribute>
+           <layout class="QVBoxLayout" name="verticalLayout_6">
+            <item>
+             <widget class="QLabel" name="runTimeText">
+              <property name="font">
```
Comment (milianw) [inline src/mainwindow.ui]: don't hardcode fonts

File: src/mainwindow.ui [hunk 15]
```diff
@@ -0,0 +1,716 @@
...
+                <property name="text">
+                 <string/>
+                </property>
+               </widget>
+              </item>
+             </layout>
+            </item>
+            <item>
+             <widget class="QLabel" name="topHotspotsText">
+              <property name="font">
```
Comment (milianw) [inline src/mainwindow.ui]: don't hardcode fonts

File: src/mainwindow.ui [hunk 16]
```diff
@@ -0,0 +1,716 @@
...
+                  <height>1</height>
+                 </size>
+                </property>
+                <property name="baseSize">
+                 <size>
+                  <width>0</width>
+                  <height>0</height>
+                 </size>
+                </property>
+                <property name="styleSheet">
```
Comment (milianw) [inline src/mainwindow.ui]: don't hardcode colors

File: src/mainwindow.ui [hunk 17]
```diff
@@ -0,0 +1,716 @@
...
+            <height>30</height>
+           </size>
+          </property>
+          <property name="maximumSize">
+           <size>
+            <width>16777215</width>
+            <height>16777215</height>
+           </size>
+          </property>
+          <property name="font">
```
Comment (milianw) [inline src/mainwindow.ui]: don't hardcode font size

**Comment by nwrogers** [comment]
Removed custom icons, fonts, and styles.

**Pull request merged**